src/cmd/troubleshoot/builder.go

```diff
@@ -95,49 +101,122 @@ func (builder CommandBuilder) buildRun() func(*cobra.Command, []string) error {
 
 		apiReader := k8scluster.GetAPIReader()
 
-		troubleshootCtx := &troubleshootContext{
+		troubleshootCtx := troubleshootContext{
 			context:       context.Background(),
 			apiReader:     apiReader,
 			httpClient:    &http.Client{},
 			namespaceName: namespaceFlagValue,
-			dynakubeName:  dynakubeFlagValue,
 		}
 
 		results := NewChecksResults()
+		_ = runChecks(results, &troubleshootCtx, getDynakubeUnrelatedChecks()) // ignore error to avoid polluting pretty logs
+		resetLogger()
+
+		dynakubes, err := getDynakubes(troubleshootCtx, dynakubeFlagValue)
+		if err != nil {
+			return nil
+		}
+
+		runChecksForAllDynakubes(results, getDynakubeSpecificChecks(results, getDynakubeUnrelatedChecks()), dynakubes, apiReader)
 
-		_ = runChecks(results, troubleshootCtx, getChecks(results)) // ignore error to avoid polluting pretty logs
 		return nil
 	}
 }
 
-func getChecks(results ChecksResults) []*Check {
+func runChecksForAllDynakubes(results ChecksResults, checks []*Check, dynakubes []dynatracev1beta1.DynaKube, apiReader client.Reader) {
+	for _, dynakube := range dynakubes {
+		results.checkResultMap = results.resetResults(getDynakubeUnrelatedChecks())
+		logNewDynakubef(dynakube.Name)
+
+		troubleshootCtx := troubleshootContext{
+			context:       context.Background(),
+			apiReader:     apiReader,
+			httpClient:    &http.Client{},
+			namespaceName: namespaceFlagValue,
+			dynakube:      dynakube,
+		}
+
+		_ = runChecks(results, &troubleshootCtx, getDynakubeSpecificChecks(results, checks)) // ignore error to avoid polluting pretty logs
+		resetLogger()
+		if !results.hasErrors() {
+			logOkf("'%s' - all checks passed", dynakube.Name)
+		}
+	}
+}
+
+func getDynakubeUnrelatedChecks() []*Check {
```

Comment:
```suggestion
func getPrerequisiteChecks() []*Check {
```

crd is very related to the Dynakubes 😅